projects/js-packages/publicize-components/src/components/form/index.js

```diff
@@ -22,41 +24,85 @@ import styles from './styles.module.scss';
  * @param {boolean} props.isPublicizeEnabled            - Whether Publicize is enabled for this post.
  * @param {boolean} props.isRePublicizeFeatureEnabled   - True if the RePublicize feature is available.
  * @param {boolean} props.isPublicizeDisabledBySitePlan - A combination of the republicize feature being enabled and/or the post not being published.
+ * @param {boolean} props.numberOfSharesRemaining       - The number of shares remaining for the current period. Optional.
  * @returns {object}                                    - Publicize form component.
  */
 export default function PublicizeForm( {
 	isPublicizeEnabled,
 	isRePublicizeFeatureEnabled,
 	isPublicizeDisabledBySitePlan,
+	numberOfSharesRemaining = null,
 } ) {
-	const { connections, toggleById, hasConnections } = useSocialMediaConnections();
+	const {
+		connections,
+		toggleById,
+		hasConnections,
+		enabledConnections,
+	} = useSocialMediaConnections();
 	const { message, updateMessage, maxLength } = useSocialMediaMessage();
 
 	const isDisabled = () =>
 		! isRePublicizeFeatureEnabled && connections.every( connection => ! connection.toggleable );
 	const Wrapper = isPublicizeDisabledBySitePlan ? Disabled : Fragment;
 
+	const outOfConnections =
+		numberOfSharesRemaining !== null && numberOfSharesRemaining <= enabledConnections.length;
+
 	return (
 		<Wrapper>
 			{ hasConnections && (
-				<PanelRow>
-					<ul className={ styles[ 'connections-list' ] }>
-						{ connections.map(
-							( { display_name, enabled, id, service_name, toggleable, profile_picture } ) => (
-								<PublicizeConnection
-									disabled={ isRePublicizeFeatureEnabled ? ! isPublicizeEnabled : ! toggleable }
-									enabled={ enabled && ! isPublicizeDisabledBySitePlan }
-									key={ id }
-									id={ id }
-									label={ display_name }
-									name={ service_name }
-									toggleConnection={ toggleById }
-									profilePicture={ profile_picture }
-								/>
-							)
-						) }
-					</ul>
-				</PanelRow>
+				<>
+					{ numberOfSharesRemaining !== null && (
+						<PanelRow>
+							<Notice type={ numberOfSharesRemaining < connections.length ? 'warning' : 'default' }>
+								<>
+									{ createInterpolateElement(
+										sprintf(
+											/* translators: %d is the number of shares remaining, upgradeLink is the link to upgrade to a different plan */
+											_n(
+												'You have %d share remaining. <upgradeLink>Upgrade</upgradeLink> to share to all your social media accounts.',
+												'You have %d shares remaining. <upgradeLink>Upgrade</upgradeLink> to share to all your social media accounts.',
+												numberOfSharesRemaining,
+												'jetpack'
+											),
+											numberOfSharesRemaining
+										),
+										{
+											upgradeLink: (
+												<a
+													href={ 'https://example.com' }
```

Comment:
Should we add the link to check out the plan here?

Reply:
I wasn't sure if we should add it yet, considering it's not really publicly available yet. What do you think?

Reply:
Makes sense, we can replace the link when we are ready to launch